helm/values.yaml

```diff
@@ -0,0 +1,110 @@
+# Default values for substrate-telemtry.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+replicaCount:
+  shard: 3
+  core: 1
+
+image:
+  repository: parity/substrate-telemetry-backend
+  pullPolicy: IfNotPresent
+  # Overrides the image tag whose default is the chart appVersion.
+  tag: latest
+
+imagePullSecrets: []
+nameOverride: ""
+fullnameOverride: ""
+
+serviceAccount:
+  # Specifies whether a service account should be created
+  create: true
+  # Annotations to add to the service account
+  annotations: {}
+  # The name of the service account to use.
+  # If not set and create is true, a name is generated using the fullname template
+  name: ""
+
+podAnnotations: {}
+
+podSecurityContext: {}
+  # fsGroup: 2000
+
+securityContext: {}
+  # capabilities:
+  #   drop:
+  #   - ALL
+  # readOnlyRootFilesystem: true
+  # runAsNonRoot: true
+  # runAsUser: 1000
+
+service:
+  shard:
+    type: LoadBalancer
+    port: 80
+    targetPort: 8000
+    annotations:
+      cloud.google.com/load-balancer-type: Internal
+      networking.gke.io/internal-load-balancer-allow-global-access: "true"
+      external-dns.alpha.kubernetes.io/hostname: substrate-telemetry.parity-stg.parity.io.
+  core:
+    type: ClusterIP
+    port: 80
+    targetPort: 8000
+    annotations: {}
+
+
+ingress:
+  enabled: false
+  className: ""
+  annotations: {}
+    # kubernetes.io/ingress.class: nginx
+    # kubernetes.io/tls-acme: "true"
+  hosts:
+    - host: chart-example.local
+      paths:
+        - path: /
+          pathType: ImplementationSpecific
+  tls: []
+  #  - secretName: chart-example-tls
+  #    hosts:
+  #      - chart-example.local
+
+resources: {}
+  # We usually recommend not to specify default resources and to leave this as a conscious
+  # choice for the user. This also increases chances charts run on environments with little
+  # resources, such as Minikube. If you do want to specify resources, uncomment the following
+  # lines, adjust them as necessary, and remove the curly braces after 'resources:'.
+  # limits:
+  #   cpu: 100m
+  #   memory: 128Mi
+  # requests:
+  #   cpu: 100m
+  #   memory: 128Mi
+
+autoscaling:
+  enabled: false
```

Comment:
I guess the first deployment will be without autoscaling enabled? I did notice the `hpa` helm file above.

Reply:
The core cannot be autoscaled. Exactly 1 instance of it can be running, since it is stateful. A "next step" in the architecture could allow multiple cores, but even then the shards would need restarting (for now, unless we did something smarter) if new cores were added because they'd need to find out the new IP address to send telemetry to.

Shards could be autoscaled up, but removing a shard (while fine) would lead to a temporary loss of telemetry for all connected nodes until they reconnected to one of the still-present shards again.

Reply:
added HorizontalPodAutoscaling to telemetry-shard deployment.